microprofile/health-smallrye/src/main/java/org/wildfly/extension/microprofile/health/deployment/CDIExtension.java

```diff
@@ -74,19 +69,22 @@
     public CDIExtension(HealthReporter healthReporter, Module module) {
         this.reporter = healthReporter;
         this.module = module;
-
     }
 
     /**
      * Get CDI <em>instances</em> of HealthCheck and
      * add them to the {@link HealthReporter}.
      */
-    private void afterDeploymentValidation(@Observes final AfterDeploymentValidation avd, BeanManager bm) {
-        instance = bm.createInstance();
-
-        addHealthChecks(HealthLiteral.INSTANCE, reporter::addHealthCheck, healthChecks);
-        addHealthChecks(Liveness.Literal.INSTANCE, reporter::addLivenessCheck, livenessChecks);
-        addHealthChecks(Readiness.Literal.INSTANCE, reporter::addReadinessCheck, readinessChecks);
+    private void afterDeploymentValidation(@Observes final AfterDeploymentValidation avd,
+                                           BeanManager eventBeanManager) {
+        // Get the bean manager from which to resolve all healthchecks in the deployment.
+        BeanManager beanManager = getBeanManager(module, eventBeanManager);
+        instance = beanManager.createInstance();
+
+        // Register them to be called using this module classloader
+        addHealthChecks(HealthLiteral.INSTANCE, reporter::addHealthCheck, module, healthChecks);
```

Comment:
module does not need to be passed to these methods. Current code is already taking the `module.getClassLoader()`